### PR TITLE
BVAL-605 Avoid empty initialize() methods in spec examples

### PR DESCRIPTION
**Pull request by gsmet**
And fix a few examples in passing.

 * https://hibernate.atlassian.net/browse/BVAL-605

### CODE REVIEW

File: sources/constraint-definition.asciidoc [hunk 1]
```diff
@@ -618,12 +618,12 @@ And some invalid definitions in <<example-constraintsdefinitionimplementation-va
 [source, JAVA]
 ----
 //parameterized type
-public class SizeValidatorForString implements<Size, Collection<String>> {
+public class SizeValidatorForString implements ConstraintValidator<Size, Collection<String>> {
```
Comment (gunnarmorling) [inline sources/constraint-definition.asciidoc]: Ooops. Good spot!

File: spec-examples/src/test/java/org/beanvalidation/specexamples/constraintdefinition/validator/DateParametersConsistentValidator.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@SupportedValidationTarget(ValidationTarget.PARAMETERS)
+public class DateParametersConsistentValidator implements ConstraintValidator<DateParametersConsistent, Object[]> {
+
+	/**
+	 * Validate a specified value. returns false if the specified value does not conform to the definition
+	 */
+	@Override
+	public boolean isValid(Object[] value, ConstraintValidatorContext context) {
+		if ( value.length != 3 ) {
+			throw new IllegalStateException( "Unexpected method signature" );
```
Comment (gunnarmorling) [inline spec-examples/src/test/java/org/beanvalidation/specexamples/constraintdefinition/validator/DateParametersConsistentValidator.java]: It's tangential, but `IllegalArgumentException` seems more reasonable.

Comment (gsmet) [inline spec-examples/src/test/java/org/beanvalidation/specexamples/constraintdefinition/validator/DateParametersConsistentValidator.java]: Fixed and force-pushed.

**Comment by gunnarmorling** [comment]
Thanks, @gsmet!

**Pull request merged**